src/generator.rs

```diff
@@ -90,45 +90,161 @@ pub fn run_generator(devices: &[Device], output_directory: &Path, fake_mode: boo
     Ok(())
 }
 
+fn write_contents(output_directory: &Path, filename: &str, contents: &str) -> Result<()> {
+    let path = output_directory.join(filename);
+    make_parent(&path)?;
+
+    let contents = format!(
+        "\
+# Automatically generated by {exe_name:?}
+
+{contents}",
+        exe_name = std::env::current_exe().unwrap(),
```

Comment:
Likewise regarding the full path here.

Also, I think doing `.display()` here and removing the `:?` above would look less weird in the general case.